web-ui/view/step.ml

```diff
@@ -563,64 +579,73 @@ module Make (M : Git_forge_intf.Forge) = struct
-                         startingLine && parseInt($el.id.substring(1, \
-                         $el.id.length)) <= endingLine ? 'highlight' : ''";
-                      Tyxml_helpers.at_click "highlightLine";
-                      a_id line_number_id;
-                    ]
-                  [
-                    td
-                      ~a:
-                        [
-                          a_class [ "code-line__number" ];
-                          a_user_data "line-number" line_number_id;
-                        ]
-                      [ txt (Printf.sprintf "%d" !line_number) ];
-                    td
-                      ~a:
-                        [
-                          a_class [ code_line_class ];
-                          a_user_data "line-number" line_number_id;
-                        ]
-                      [
-                        pre
-                          ~a:[ a_user_data "line-number" line_number_id ]
-                          [ Unsafe.data log_line ];
-                      ];
-                  ]))
+          let line =
+            Fmt.str "%a" (pp_elt ())
+              (span
+                 ~a:
+                   [
+                     a_class [ "tr" ];
+                     Tyxml_helpers.colon_class
+                       "parseInt($el.id.substring(1, $el.id.length)) >= \
+                        startingLine && parseInt($el.id.substring(1, \
+                        $el.id.length)) <= endingLine ? 'highlight' : ''";
+                     Tyxml_helpers.at_click "highlightLine";
+                     a_id line_number_id;
+                   ]
+                 [
+                   span
+                     ~a:
+                       [
+                         a_class [ "th" ];
+                         a_user_data "line-number" line_number_id;
+                       ]
+                     [];
+                   code
+                     ~a:
+                       [
+                         a_class [ code_line_class ];
+                         a_user_data "line-number" line_number_id;
+                       ]
+                     [ Unsafe.data log_line ];
+                 ])
+          in
+          Some line
       in
-      Printf.sprintf "%s%s"
-        (List.fold_left aux
-           "<table data-paste-markdown-skip class='flex steps-table fg-default \
-            bg-default'><tbody class=\"bg-default\">"
-           data)
-        "</tbody></table>"
+      List.filter_map aux data |> String.concat "\n"
+    in
+    let collapse_carriage_returns log_line =
+      let rec last = function
+        | [] -> raise (Failure "Trying to take log_line from empty list (BUG)")
+        | s :: [] -> s
+        | _ :: l -> last l
+      in
+      match log_line with
+      | "" -> ""
+      | log_line ->
+          Dream.log "%S" log_line;
+          let split = Astring.String.cuts ~sep:"\r" log_line in
+          if List.length split > 1 then List.iter (Dream.log "%S") split;
```

Comment:
```suggestion
          if (List.compare_length_with split 1) >= 1 then List.iter (Dream.log "%S") split;
```
`List.compare_length_with` stops the computation at 1 iteration. It avoids unecessary computation.